src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala

```diff
@@ -1208,4 +1240,233 @@ class ProcessApiServiceImpl(
         complete(errorResponse.status, errorResponse)
     }
   }
+
+  override def onboardingApprove(
+    tokenId: String
+  )(implicit toEntityMarshallerProblem: ToEntityMarshaller[Problem], contexts: Seq[(String, String)]): Route = {
+    logger.info(s"Onboarding Approve having tokenId $tokenId")
+
+    def createUser(relationships: Seq[Relationship], institutionInternalId: String, u: UserRegistryUser) = {
+      // val email = u.email.filter(i => i == institutionInternalId)
+      val email = u.email match {
+        case Some(value) => value.getOrElse(institutionInternalId, "")
+        case _           => ""
+      }
+      /* val email = u.email.get.map {
+        case (institutionInternalId, str1) => str1
+        case _                             => ""
+      }*/
+
+      User(
+        id = u.id,
+        taxCode = u.taxCode.getOrElse(""),
+        name = u.name.getOrElse(""),
+        surname = u.surname.getOrElse(""),
+        email = Option(email),
+        role = PartyRoleAPIConverter.fromClientValue(relationships.filter(_.from == u.id).head.role.toString),
+        productRole = relationships.filter(_.from == u.id).head.product.role
+      )
+    }
+
+    val result: Future[Unit] = for {
+      bearer      <- getFutureBearer(contexts)
+      tokenIdUUID <- tokenId.toFutureUUID
+      token       <- partyManagementService.verifyToken(tokenIdUUID)
+      uid         <- getUidFuture(contexts)
+      userId      <- uid.toFutureUUID
+      currentUser <- userRegistryManagementService.getUserById(userId)
+
+      managerLegals = token.legals.filter(_.role == PartyManagementDependency.PartyRole.MANAGER)
+
+      managerRegistryUsers <- Future.traverse(managerLegals)(legal =>
+        userRegistryManagementService.getUserWithEmailById(legal.partyId)
+      )
+      managerRelationships <- Future.traverse(managerLegals)(relationship =>
+        partyManagementService.getRelationshipById(relationship.relationshipId)(bearer)
+      )
+      institutions         <- Future.traverse(managerRelationships)(legalUser =>
+        partyManagementService.retrieveInstitution(legalUser.to)(bearer)
+      )
+
+      institutionInternalId = institutions.headOption.map(_.id.toString).getOrElse("")
+
+      productId    = managerRelationships.head.product.id
+      managerUsers = managerRegistryUsers.map(u => createUser(managerRelationships, institutionInternalId, u))
+
+      manager             = managerRegistryUsers.head;
```

Comment:
⚠️ `head` throws an exception if the list is empty

Reply:
Added a check before that can throw ManagerFoundError before use head

Reply:
using `head` and `get` are considered bad practices in Scala, in this specific case you should prefer something like 
```
manager <- managerRegistryUsers.headOption.toFuture(yourException)
```

same for row 1293